client/java/src/main/java/io/openlineage/client/utils/ThreadFactory.java

```diff
@@ -0,0 +1,52 @@
+/*
+/* Copyright 2018-2025 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+
+package io.openlineage.client.utils;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * A ThreadFactory implementation that creates named threads for use in ThreadPoolExecutor. This
+ * factory creates threads with a specified name prefix followed by a sequential number.
+ */
+public class ThreadFactory implements java.util.concurrent.ThreadFactory {
```

Comment:
I've done some other executor service related changes in other PR -> https://github.com/OpenLineage/OpenLineage/pull/3779

I like the thread naming idea, but:
 * I would love to have a single ExecutorService for all OL related threads
 * Close the service in OpenLineage.close method
 
I think we should avoid public `ThreadFactory` class.

Reply:
@mobuchowski do you agree that there should be a single `ExecutorService` for all the OpenLineage related threads? 

If so, perhaps you could add some singleton and avoid `public` ThreadFactory. 
Passing the `namePrefix` to the thread factory does not make sense for a single `ExecutorService` either.